kornia/core/module.py

```diff
@@ -0,0 +1,275 @@
+        raise TypeError("Input type not supported")
+
+    def to_numpy(self, x: Any) -> np.array:  # type: ignore
+        """Convert input to numpy array.
+
+        Args:
+            x: The input to convert.
+
+        Returns:
+            np.array: The converted numpy array.
+        """
+        if isinstance(x, (Tensor,)):
+            return x.cpu().detach().numpy()
+        if isinstance(x, (np.ndarray,)):  # type: ignore
+            return x
+        if isinstance(x, (Image.Image,)):  # type: ignore
+            return np.array(x)  # type: ignore
+        raise TypeError("Input type not supported")
+
+    def to_pil(self, x: Any) -> Image.Image:  # type: ignore
+        """Convert input to PIL image.
+
+        Args:
+            x: The input to convert.
+
+        Returns:
+            Image.Image: The converted PIL image.
+        """
+        if isinstance(x, (Tensor,)):
+            x = x.cpu().detach() * 255
+            if x.dim() == 3:
+                x = x.permute(1, 2, 0)
+                return Image.fromarray(x.byte().numpy())  # type: ignore
+            elif x.dim() == 4:
+                x = x.permute(0, 2, 3, 1)
+                return [Image.fromarray(_x.byte().numpy()) for _x in x]  # type: ignore
+            else:
+                raise NotImplementedError
+        if isinstance(x, (np.ndarray,)):  # type: ignore
+            raise NotImplementedError
+        if isinstance(x, (Image.Image,)):  # type: ignore
+            return x
+        raise TypeError("Input type not supported")
+
+    def _detach_tensor_to_cpu(
+        self, output_image: Union[Tensor, List[Tensor], Tuple[Tensor]]
+    ) -> Union[Tensor, List[Tensor], Tuple[Tensor]]:
+        if isinstance(output_image, (Tensor,)):
+            return output_image.detach().cpu()
+        if isinstance(
+            output_image,
+            (
+                list,
+                tuple,
+            ),
+        ):
+            return type(output_image)([self._detach_tensor_to_cpu(out) for out in output_image])  # type: ignore
+        raise RuntimeError(f"Unexpected object {output_image} with a type of `{type(output_image)}`")
+
+    def show(self, n_row: Optional[int] = None, backend: str = "pil", display: bool = True) -> Optional[Any]:
+        """Returns PIL images.
+
+        Args:
+            n_row: Number of images displayed in each row of the grid.
+            backend: visualization backend. Only PIL is supported now.
+        """
+        if self._output_image is None:
+            raise ValueError("No pre-computed images found. Needs to execute first.")
+
+        if len(self._output_image.shape) == 3:
+            out_image = self._output_image
+        elif len(self._output_image.shape) == 4:
+            if n_row is None:
+                n_row = math.ceil(self._output_image.shape[0] ** 0.5)
+            out_image = kornia.utils.image.make_grid(self._output_image, n_row, padding=2)
+        else:
+            raise ValueError
+
```

Comment:
Shouldn't the terminal print you implemented some time ago be supported here?

Reply:
Yes, maybe.

Reply:
naming this method `show` is a bit confusing

Reply:
why? any other names?